quacc/custodian/qchem.py

```diff
@@ -0,0 +1,76 @@
+"""
+Custodian handlers for QChem
+"""
+from __future__ import annotations
+
+import os
+import shlex
+
+from custodian import Custodian
+from quacc import SETTINGS
+
+
+def run_custodian(
+    qchem_cmd: str = SETTINGS.QChem_CMD,
+    qchem_max_cores: int = SETTINGS.QChem_MAX_CORES,
+    qchem_calc_loc: str = SETTINGS.QChem_CALC_LOC,
+    qchem_custodian_max_errors: int = SETTINGS.QChem_CUSTODIAN_MAX_ERRORS,
+    qchem_custodian_handlers: list[str] = SETTINGS.QChem_CUSTODIAN_HANDLERS,
+) -> None:
+    """
+    Function to run QChem Custodian
+
+    Parameters
+    ----------
+    qchem_cmd
+        Q-Chem command. Defaults to "qchem" in settings.
+    qchem_max_cores
+        Maximum number of cores to use for the Q-Chem calculation. Defaults to 32 in settings.
+    qchem_calc_loc
+        Compute-node local scratch directory in which Q-Chem should perform IO. Defaults to /tmp in settings.
+    qchem_custodian_max_errors
+        Maximum number of errors to allow before stopping the run. Defaults to 5 in settings.
+    qchem_custodian_handlers
+        List of handlers to use in Custodian. See settings for list.
+
+    Returns
+    -------
+    None
+    """
+    # Adapted from atomate.qchem.firetasks.run_calc
+
+    from custodian.qchem.handlers import QChemErrorHandler
+    from custodian.qchem.jobs import QCJob
+
+    # Handlers for Q-Chem
+    handlers_dict = {
+        "QChemErrorHandler": QChemErrorHandler(),
+    }
+
+    handlers = []
+    for handler_flag in qchem_custodian_handlers:
+        if handler_flag not in handlers_dict:
+            raise ValueError(f"Unknown Q-Chem error handler: {handler_flag}")
+        handlers.append(handlers_dict[handler_flag])
+
+    # Run Q-Chem
+
+    jobs = [
+        QCJob(
+            qchem_command=qchem_cmd,
+            max_cores=qchem_max_cores,
+            calc_loc=qchem_calc_loc,
```

Comment:
Should we do a check like `if not os.path.exists(qchem_calc_loc)` then `raise ValueError()` to make sure the path exists, or is that checked upstream in `QCJob`?

Reply:
Sure, a check like this makes sense. The one thing to keep in mind is that the chem_calc_loc path (which defaults to /tmp) will most likely only exist on the compute node.

Reply:
Let's rename this to `qchem_scratch_loc` or something similar to be better aligned with the `QCLOCALSCR` environment variable that it inherits from in custodian if set to `None`.

Reply:
I think that naming it `qchem_local_scratch_dir` in quacc is fine. I'm a bit hesitant to have that be set by `SCRATCH_DIR` because of the potential confusion between Q-Chem's two different scratch directories - `QCLOCALSCR` and `QCSCRATCH`.